### PR TITLE
CRM-19197 merge links fix and CRM-19198 Add form merge hook call

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
- [CRM-19197: Stop Merge links from dropping the limit parameter](https://issues.civicrm.org/jira/browse/CRM-19197)
  - [CRM-19198: Add form merge hook call to dedupe form](https://issues.civicrm.org/jira/browse/CRM-19198)

### CODE REVIEW

**Comment by seamuslee001** [comment]
I haven't tested this yet but looking over the code this looks good will try and test in the next day or so

File: CRM/Contact/Page/DedupeFind.php [hunk 1]
```diff
@@ -35,8 +35,9 @@ class CRM_Contact_Page_DedupeFind extends CRM_Core_Page_Basic {
   protected $_rgid;
   protected $_mainContacts;
   protected $_gid;
+  protected $action;
 
-  /**
+	/**
```
Comment (seamuslee001) [inline CRM/Contact/Page/DedupeFind.php]: um @eileenmcnaughton ?

Comment (eileenmcnaughton) [inline CRM/Contact/Page/DedupeFind.php]: dang

File: CRM/Contact/Form/Merge.php [hunk 2]
```diff
@@ -321,21 +317,21 @@ public function postProcess() {
...
+    $migrationData = array('migration_info' => $formValues);
+    CRM_Utils_Hook::merge('form', $migrationData, $this->_cid, $this->_oid);
+    CRM_Dedupe_Merger::moveAllBelongings($this->_cid, $this->_oid, $migrationData['migration_info']);
 
     $name = CRM_Core_DAO::getFieldValue('CRM_Contact_DAO_Contact', $this->_cid, 'display_name');
     $message = '<ul><li>' . ts('%1 has been updated.', array(1 => $name)) . '</li><li>' . ts('Contact ID %1 has been deleted.', array(1 => $this->_oid)) . '</li></ul>';
     CRM_Core_Session::setStatus($message, ts('Contacts Merged'), 'success');
 
     $url = CRM_Utils_System::url('civicrm/contact/view', "reset=1&cid={$this->_cid}");
+    $urlParams = "reset=1&gid={$this->_gid}&rgid={$this->_rgid}&limit={$this->limit}";
```
Comment (seamuslee001) [inline CRM/Contact/Form/Merge.php]: @eileenmcnaughton if limit is not in the url it just has in the url &limit=& which seems odd

Comment (eileenmcnaughton) [inline CRM/Contact/Form/Merge.php]: Yeah - but fundamentally harmless - it seems like we have lots of if this exists then add it to the url which seems like a lot of code complexity for no gain

**Comment by eileenmcnaughton** [comment]
@seamuslee001 did you get to testing this? I made one more tiny fix (rebased into it)

**Comment by seamuslee001** [comment]
@eileenmcnaughton I tested this on our test site, It appeared as far as i can tell to not break anything so i think this is good, I didn't test out trying to interact with the hook tho

**Comment by eileenmcnaughton** [comment]
Thanks @seamuslee001

**Pull request merged**